src/fparser/tests/test_readfortran.py

```diff
@@ -390,9 +396,23 @@ def test_file_reader():
         raise
 
 
+##############################################################################
+
+def test_bad_file_reader():
+    '''
+    Tests that the file reader can spot when it is given something to read
+    which is neither file nor filename.
+    '''
+    with pytest.raises(Exception):
+        unit_under_test = FortranFileReader(42)
```

Comment:
Please check that this raises the Exception we are looking for, e.g.:

    with pytest.raises(Exception) as err:
        blah
    assert "your error msg here" in str(err)

I think this is causing an error earlier than the one we want and thus lines 1375-1377 are still not covered.

Reply:
You were right. The call to `fparser.sourceinfo.get_source_info` was throwing an exception because it too checks that it has been passed the correct type of argument. I can't see any reason to do it first so I've moved it after.

Reply:
Good plan.